applications/GeoMechanicsApplication/custom_processes/set_parameter_field_process.cpp

```diff
@@ -0,0 +1,143 @@
+
+#include <utilities/function_parser_utility.h>
+#include <utilities/mortar_utilities.h>
+
+#include "utilities/interval_utility.h"
+#include "geo_mechanics_application_variables.h"
+
+
+namespace Kratos
+{
+
+    SetParameterFieldProcess::SetParameterFieldProcess(ModelPart& rModelPart,
+                                                            Parameters Settings)
+                                                            : mrModelPart(rModelPart),
+                                                            mParameters(Settings)
+{
+    // function type: python, cpp, input
+    Parameters default_parameters(R"(
+        {
+            "help"            : "This process applies a moving load condition belonging to a modelpart. The load moves over line elements.",
+            "model_part_name" : "please_specify_model_part_name",
+            "variable_name"   : "CUSTOM",
+            "func_type"       : "input",               
+            "function"        : "0",
+            "dataset"         : "dummy"
+        }  )"
+    );
+    Parameters mParameters;
+
+    mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+}
+
+
+void SetParameterFieldProcess::SetValueAtElement(Element& rElement, const Variable<double>& rVar, double Value)
+{
+
+
+    Properties& r_prop = rElement.GetProperties();
+
+    // Copies properties
+    Properties::Pointer p_new_prop = Kratos::make_shared<Properties>(r_prop);
+
+    // Adds new properties to the element
+    p_new_prop->SetValue(rVar, Value);
+    rElement.SetProperties(p_new_prop);
+
+}
+
+
+void SetParameterFieldProcess::ExecuteInitialize()
+{
+    KRATOS_TRY
+
+    if (!this->mrModelPart.GetProcessInfo()[IS_RESTARTED]){
+
+        const Variable<double>& r_var = KratosComponents< Variable<double> >::Get(mParameters["variable_name"].GetString());
+
+        // set parameter field from input function
+        if (mParameters["func_type"].GetString().compare("input") == 0)
+        {
+
+            BasicGenericFunctionUtility parameter_function = BasicGenericFunctionUtility(mParameters["function"].GetString());
+
+            const double current_time = this->mrModelPart.GetProcessInfo().GetValue(TIME);
+
+            for (Element& r_element : mrModelPart.Elements()) {
+
+                auto& r_geom = r_element.GetGeometry();
+
+                // calculate parameter value at current element
+                const double val = parameter_function.CallFunction(r_geom.Center().X(), r_geom.Center().Y(), r_geom.Center().Z(), current_time, 0, 0, 0);
+                
+
+                this->SetValueAtElement(r_element, r_var, val);
+
+            }
+        }
+        // set parameter field with a python function
+        else if (mParameters["func_type"].GetString().compare("python") == 0)
+        {
+
+            // get new data from the data set
+            const std::string dataset = mParameters["dataset"].GetString();
+            Parameters new_data = Parameters(dataset);
```

Comment:
No need to use the copy constructor here. The simplest way to achieve what you want is probably this:
```suggestion
            Parameters new_data{dataset};
```